keps/sig-scheduling/20200612-deducted-topology-manager.md

```diff
@@ -0,0 +1,177 @@
+---
+title: Deducted version of TopologyManager in kube-scheduler
+authors:
+  - "@AlexeyPerevalov"
+owning-sig: sig-scheduling
+participating-sigs:
+reviewers:
+  - "@huang-wei"
+approvers:
+  - "@huang-wei"
+editor: TBD
+creation-date: 2020-05-28
+last-updated: 2020-05-28
+status: implementable
+see-also:
+superseded-by:
+---
+# Topology Aware Scheduling
+
+<!-- toc -->
+- [Summary](#Summary)
+- [Motivation](#Motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#Proposal)
+- [Use cases](#use-cases)
+- [Test plans](#test-plans)
+- [Graduation criteria](#graduation-criteria)
+- [Implementation history](#implementation-history)
+<!-- /toc -->
+
+# Summary
+
+This document describes behavior of the Kubernetes Scheduler which take into
+account worker node NUMA topology.
+
+# Motivation
+
+After Topology Manager was introduced the problem of launching pod in the
+cluster where worker nodes have different NUMA topology and different amount
+of resources in that topology became actual. Pod could be scheduled on the node
+where total amount of resources are enough, but resource distribution could not
+satisfy the appropriate Topology policy. In this case the pod failed to start. Much
+better behaviour for scheduler would be to select appropriate node where admit
+handlers may pass.
+
+
+## Goals
+
+-   Make scheduling process more precise when we have NUMA topology on the
+worker node.
+
+## Non-Goals
+
+-   Do not change Topology Manager behaviour to be able to work with policy in
+the PodSpec
+-   This Proposal requires exposing NUMA topology information. This KEP doesn't
+describe how to expose all necessary information it just declare what kind of
+information is necessary.
+
+# Proposal
+Kube-scheduler builtin plugin will be added to the main tree. This plugin
+implements a deducted version of TopologyManager it’s different from original topology manager algorithm.
+Plugin checks the ability to run pod only in case of single-numa-node policy on the
+node, since it is the most strict policy, it implies that the launch on the node with
+other existing policies will be successful if the condition for single-numa-node policy passed for the worker node.
+Proposed plugin will use node label to identify which topology policy is
+enabled on the node.
+To work, this plugin requires topology information of the available resource on the worker nodes.
+
+
+## Node labels
+
+Node label contains the name of the topology policy currently implemented in kubelet.
+
+Proposed Node Label may look like this:
+  `beta.kubernetes.io/topology=none|best-effort|restricted|single-numa-node`
```

Comment:
this will be confusing, we already have ```topology.kubernetes.io```: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/well_known_labels.go#L22:1

We should run this by sig API Machinery to define a label.

Reply:
also, why do we need this label at all? can't we have the policy in the CRD?

Reply:
CRD describes node, so yes it's good idea to keep it there.

Reply:
+1 to make this info described by CRD.